test/e2e_federation/util.go

```diff
@@ -141,21 +141,84 @@ func createService(clientset *fedclientset.Clientset, namespace, name string) (*
 	}
 
 	By(fmt.Sprintf("Trying to create service %q in namespace %q", service.Name, namespace))
-	return clientset.Services(namespace).Create(service)
+	var err error
```

Comment:
I updated the createService function as you proposed: create federated service, then check creation.